engine/src/main/java/org/terasology/engine/world/block/BlockArea.java

```diff
@@ -354,4 +355,24 @@ public int hashCode() {
     public String toString() {
         return "BlockArea[(" + this.minX + ", " + this.minY + ")...(" + this.maxX + ", " + this.maxY + ")]";
     }
+
+    //@Override ->Uncomment after editing interface BlockAreac
+    public long distanceSquared(Vector2ic p){
+        return distanceSquared(p.x(), p.y());
+    }
+
+    //@Override ->Uncomment after editing interface BlockAreac
+    public long distanceSquared(int px, int py){
+        //center = min + width / 2 = min + (max - min) / 2
+        // 2 * center  = 2 * min + max - min = min + max
+        int cx2 = minX() + maxX();
+        int cy2 = minY() + maxY();
+
+        //     dx = p - center - width / 2
+        // 2 * dx = 2 * px - 2 * center - width
+        //     dx = (2 * px - 2 * center - width) / 2
+        long dx = Math.max(Math.abs(2 * px - cx2) - getSizeX(), 0) / 2;
+        long dy = Math.max(Math.abs(2 * py - cy2) - getSizeY(), 0) / 2;
+        return dx * dx + dy * dy;
+    }
```

Comment:
this looks good sorry, for the late response. so this is the square distance from the center of a block to another center. I guess do we want it to be calculated for the edge instead?

@skaldarnar

Reply:
Yes, this should be available for the immutable interface in `BlockAreac` as well.